### ISSUE TITLE
Enabling File Content Definition at a fresh installation on OC 1.4 hides Content Parts

### ISSUE THREAD

**Issue by dodyg** [OrchardCMS/OrchardCore#12629]
### Describe the bug
- I start with a blank template
- When I enable File Content Definition Module, the Content Part list is missing

![content-part](https://user-images.githubusercontent.com/1761249/195346297-1ce3a132-ac00-4d0c-aaa9-d29527c6c495.png)

**Comment by dodyg** [comment]
When I disable it

![file-content-definition](https://user-images.githubusercontent.com/1761249/195346458-796ea55b-cb75-4064-b4bd-66776ff3ae3f.png)

![available](https://user-images.githubusercontent.com/1761249/195346596-1bc752ee-6475-4769-84c4-fc76d746270d.png)

**Comment by MikeAlhayek** [comment]
When you setup a fresh site, the content types/parts are stored into the database. Then when you enable `File Content Definition` it starts looking for an empty file.

You could start with a blank recipe that depends on `File Content Definition` instead of the database.

**Comment by dodyg** [comment]
There is no blank recipe that starts with `File Content Definition`. With this situation, it makes the module pretty much useless.

**Comment by MikeAlhayek** [comment]
What I mean as a workaround, you could create your own [Blank Site](
https://github.com/OrchardCMS/OrchardCore/blob/bbe1b14fa448b9653f0523bdc40267af77489325/src/OrchardCore.Themes/TheAdmin/Recipes/blank.recipe.json) recipe that would enable `OrchardCore.Contents.FileContentDefinition` by default so the content definitions are stored in the files by default instead of database.

We could add handler in the `File Content Definition` that would merge definitions found in the database with definitions found in the database when this feature is being enabled `Enabling(Feature feature)` event. This way we combine the two sources each time this feature is enabled. Feel free to submit a PR for this if you want.
```
{
  "name": "Blank",
  "displayName": "Blank site",
  "description": "Creates a blank site with content management features.",
  "author": "The Orchard Core Team",
  "website": "https://orchardcore.net",
  "version": "1.0.0",
  "issetuprecipe": true,
  "categories": [ "default" ],
  "tags": [ "blank" ],

  // The variables are evaluated the first time they are accessed, and reused across steps
  "variables": {
    // "now": "[js: new Date().toISOString()]"
  },

  "steps": [
    {
      "name": "feature",
      "enable": [
        // SaaS
        "OrchardCore.HomeRoute",
        "OrchardCore.Admin",
        "OrchardCore.Diagnostics",
        "OrchardCore.DynamicCache",
        "OrchardCore.Features",
        "OrchardCore.Navigation",
        "OrchardCore.Recipes",
        "OrchardCore.Resources",
        "OrchardCore.Roles",
        "OrchardCore.Security",
        "OrchardCore.Settings",
        "OrchardCore.Themes",
        "OrchardCore.Users",

        // Content Management
        "OrchardCore.Contents.FileContentDefinition", /// <<<<< THIS WAS ADDED
        "OrchardCore.Alias",
        "OrchardCore.Autoroute",
        "OrchardCore.Html",
        "OrchardCore.ContentFields",
        "OrchardCore.ContentPreview",
        "OrchardCore.Contents",
        "OrchardCore.ContentTypes",
        "OrchardCore.CustomSettings",
        "OrchardCore.Deployment",
        "OrchardCore.Deployment.Remote",
        "OrchardCore.Flows",
        "OrchardCore.Indexing",
        "OrchardCore.Layers",
        "OrchardCore.Lists",
        "OrchardCore.Markdown",
        "OrchardCore.Media",
        "OrchardCore.Menu",
        "OrchardCore.Queries",
        "OrchardCore.Shortcodes.Templates",
        "OrchardCore.Title",
        "OrchardCore.Templates",
        "OrchardCore.Widgets",

        // Themes
        "TheAdmin"
      ]
    },
    {
      "name": "themes",
      "admin": "TheAdmin",
      "site": ""
    }
  ]
}
```

**Comment by dodyg** [comment]
> This isn't a bug but the expected behavior. When the site is first setup, the content definitions are stored in the database. But when you enable File Content Definition the content definition source is now an empty file with no definitions.

The thing is, right now the act of enabling and disabling `OrchardCore.Contents.FileContentDefinition` module make users 'lose' data. The system instantly become unusable because all the Content Parts, etc are missing. 

So perhaps technically the module does what it is supposed to do, but from a user's perspective, this module is a minefield.

**Comment by MikeAlhayek** [comment]
@dodyg yes I understand. That is why I suggest adding a feature handler in `OrchardCore.Contents.FileContentDefinition` that would copy the existing definitions from the database into the file `OrchardCore.Contents.FileContentDefinition`. And if disabling `OrchardCore.Contents.FileContentDefinition`, then again copy the data from the file back into the database.  This way, `ContentDefinitionManager` has a copy of the most recent content definitions

**Comment by Piedone** [comment]
You need to do this in reverse: https://docs.orchardcore.net/en/latest/docs/guides/content-definitions/ Maybe we should update the documentation and display a warning when switching content definition stores. I wouldn't implement automatic migration, since that looks like a lot of work for something that at most happens once for everyone, and migration with Deployment is available.

**Comment by dodyg** [comment]
The safest way is to simply to remove the module from standard installation and not give people the option.

**Comment by Skrypt** [comment]
I believe that the module has been removed from the standard setup recipes. Now, I don't think making this automatic when enabling/disabling the feature will do any good since it could create more confusion and also would eventually lead to another issue which is that it could potentially overwrite any changes that was made previously in the DB or File Content Definition if someone would toggle the feature in the admin UI. So, the best solution is adding documentation about the feature.
